### PR TITLE
Bug/1682/wrong reset button position

### PR DESCRIPTION
**Pull request by RomanenkoVladimir**
# Wrong reset button position for top bar components

_**Please read the [CONTRIBUTING.md](https://github.com/MaibornWolff/codecharta/blob/main/CONTRIBUTING.md) before opening a PR.**_

closes: #1682 

## Description

- fix issue with incorrect reset button position
- fix small styling issue with color picker buttons
- add e2e test for reset button to assure it wont get moved in the future

## Screenshots or gifs

fixedResetIcon

### CODE REVIEW

File: visualization/app/codeCharta/ui/ribbonBar/ribbonBar.e2e.ts [hunk 1]
```diff
@@ -121,9 +122,29 @@ describe("RibbonBar", () => {
...
 	})
+
+	it("should check if reset button height matches the height of the last entry of color component", async () => {
+		const colorPanel = "color-metric"
+
+		await ribbonBar.togglePanel(colorPanel)
+		const boundingBoxCheckbox = await ColorSettingsPageObject.toggleInverColorBoundingBox()
+		const boundingBoxResetButton = await ColorSettingsPageObject.resetButtonBoundingBox()
+
+		expect(Math.round(boundingBoxCheckbox.y)).toEqual(Math.floor(boundingBoxResetButton.y))
```
Comment (Med-H) [inline visualization/app/codeCharta/ui/ribbonBar/ribbonBar.e2e.ts:138]: without trying it, I'm not sure about this equality check

File: visualization/app/codeCharta/ui/ribbonBar/ribbonBar.e2e.ts [hunk 2]
```diff
@@ -121,9 +122,29 @@ describe("RibbonBar", () => {
...
+	})
+
+	it("should check if reset button height matches the height of the last entry of area component", async () => {
+		const areaPanel = "area-metric"
+
+		await ribbonBar.togglePanel(areaPanel)
+		const boundingBoxCheckbox = await AreaSettingsPanelPageObject.toggleMarginBoundingBox()
+		const boundingBoxResetButton = await AreaSettingsPanelPageObject.resetButtonBoundingBox()
+
+		expect(Math.floor(boundingBoxCheckbox.y)).toEqual(Math.floor(boundingBoxResetButton.y))
```
Comment (Med-H) [inline visualization/app/codeCharta/ui/ribbonBar/ribbonBar.e2e.ts:148]: same as above.

File: visualization/app/codeCharta/ui/ribbonBar/ribbonBar.e2e.ts (hunk 1 above)
Comment (RomanenkoVladimir) [reply visualization/app/codeCharta/ui/ribbonBar/ribbonBar.e2e.ts:138]: Yes, it seems to have a variance depending on the browser, I should increase the tolerance window, the idea is to check if the coordinates roughly match. +-5 px will not affect the component to much, but 150 like in the issue is a big problem.

Comment (RomanenkoVladimir) [reply visualization/app/codeCharta/ui/ribbonBar/ribbonBar.e2e.ts:138]: Thanks I will change it

**Comment by BridgeAR** [comment]
Would it be possible to have the same space between the entries? :-) Especially the space between the color and the text next to it seems to narrow, compared to the other entries.

File: visualization/app/codeCharta/ui/edgeSettingsPanel/edgeSettingsPanel.component.html [hunk 3]
```diff
@@ -64,14 +64,14 @@
...
-<md-input-container class="md-block">
+<md-input-container>
 	<md-checkbox
 		class="md-primary"
 		ng-model="$ctrl._viewModel.showOnlyBuildingsWithEdges"
 		ng-change="$ctrl.applyShowOnlyBuildingsWithEdges()"
 		ng-disabled="$ctrl._viewModel.totalAffectedBuildings == 0"
 	>
-		Show only nodes with edges
+		Only nodes with edges
```
Comment (ce-bo) [inline visualization/app/codeCharta/ui/edgeSettingsPanel/edgeSettingsPanel.component.html:74]: NIT: I personally prefer the old style.

File: visualization/app/codeCharta/ui/ribbonBar/ribbonBar.component.scss [hunk 4]
```diff
@@ -118,7 +118,7 @@ ribbon-bar-component {
 				}
 
 				reset-settings-button-component .md-button {
-					top: 0;
+					top: -7.5px;
```
Comment (ce-bo) [inline visualization/app/codeCharta/ui/ribbonBar/ribbonBar.component.scss:121]: This is not very nice - but ok :)

**Review by ce-bo — APPROVED** [review]
LGTM

File: visualization/app/codeCharta/ui/edgeSettingsPanel/edgeSettingsPanel.component.html (hunk 3 above)
Comment (RomanenkoVladimir) [reply visualization/app/codeCharta/ui/edgeSettingsPanel/edgeSettingsPanel.component.html:74]: It didnt fit in 1 line when trying to display the icon next to it, and making a line break for the icon didn't look really nice

File: visualization/app/codeCharta/ui/ribbonBar/ribbonBar.component.scss (hunk 4 above)
Comment (RomanenkoVladimir) [reply visualization/app/codeCharta/ui/ribbonBar/ribbonBar.component.scss:121]: How would you prefer to do it?

**Comment by Med-H** [comment]
What was LGTM again ?

**Comment by RomanenkoVladimir** [comment]
Looks good to me

**Pull request merged**